ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/attestation/AggregatingAttestationPool.java

```diff
@@ -73,6 +76,16 @@ public synchronized void remove(final Attestation attestation) {
   }
 
   private boolean canBeIncluded(final MatchingDataAttestationGroup group, final UnsignedLong slot) {
-    return group.getAttestationData().getEarliestSlotForProcessing().compareTo(slot) <= 0;
+    final AttestationData attestationData = group.getAttestationData();
+    return attestationData.getEarliestSlotForProcessing().compareTo(slot) <= 0
+        && isPreviousEpochOrLater(attestationData, slot);
+  }
+
+  private boolean isPreviousEpochOrLater(
+      final AttestationData attestationData, final UnsignedLong slot) {
+    return compute_epoch_at_slot(attestationData.getSlot())
```

Comment:
Why do you need to do epoch + 1 here?

Reply:
The previous epoch is allowed, so if epoch of `attestation + 1 >= epoch of slot` it's ok.  The more intuitive approach would have been if `epoch of attestation >= epoch of slot - 1` but then you have to deal with the genesis epoch specially.

Reply:
Makes sense.